moped-editor/src/utils/activityLogFormatters/mopedFundingActivity.js

```diff
@@ -0,0 +1,42 @@
+import MonetizationOnOutlinedIcon from "@material-ui/icons/MonetizationOnOutlined";
+import DeleteOutlineIcon from "@material-ui/icons/DeleteOutline";
+import { ProjectActivityLogTableMaps } from "../../views/projects/projectView/ProjectActivityLogTableMaps";
+
+export const formatFundingActivity = (change) => {
+  const entryMap = ProjectActivityLogTableMaps["moped_proj_funding"];
+
+  let changeIcon = <MonetizationOnOutlinedIcon />;
+  let changeDescription = "Project funding updated";
+  let changeValue = "";
+
+  // add a new funding source
+  if (change.description.length === 0) {
+    changeDescription = "Added a new funding source ";
+    return { changeIcon, changeDescription, changeValue };
+  }
+
+  // delete existing record
+  if (change.description[0].field === "is_deleted") {
+    changeIcon = <DeleteOutlineIcon />;
+    changeDescription = "Funding source was deleted";
+    return { changeIcon, changeDescription, changeValue };
+  }
+
+  // Multiple fields in the moped_proj_funding table can be updated at once
+  // We list the fields changed in the activity log, this gathers the fields changed
+  const newRecord = change.record_data.event.data.new;
+  const oldRecord = change.record_data.event.data.old;
+
+  let changes = [];
+
+  Object.keys(newRecord).forEach((field) => {
+    if (newRecord[field] !== oldRecord[field]) {
+      changes.push(entryMap.fields[field].label);
+    }
+  });
```

Comment:
i'm getting some inconsistent results when editing funding records. if i edit an amount, the change log says
```
Edited a funding source fund, department unit, funding amount
```

when i edited a program, it says

```
Edited a funding source fund, department unit, funding amount, funding program
```

i think part of this is that we're setting amount to `0` instead of leaving it `null`....

Reply:
the other part was that fund & dept units are objects.

im bringing in a lodash function to compare object equality, lemme know if you prefer us to roll our own utility function. we use some other lodash functions elsewhere....

Reply:
Lodash's 36 million weekly downloads (!!) from npmjs.com makes me feel that you bet on the right library here. 🙌

![image](https://user-images.githubusercontent.com/3653206/213056636-0757b62e-e426-46a9-9704-9d89025789da.png)